loopy/statistics.py

```diff
@@ -1182,180 +1357,155 @@ def get_iname_strides(tag_to_iname_dict):
-
-                count_map[MemAccess(
-                        mtype="local",
-                        dtype=dtype,
-                        lid_strides=dict(sorted(lid_strides.items())),
-                        gid_strides=dict(sorted(gid_strides.items())),
-                        variable=name,
-                        count_granularity=self.local_mem_count_granularity,
-                        kernel_name=self.knl.name)] = self.one
-
-        return self.new_poly_map(count_map)
-
-    def map_variable(self, expr):
-        return self.count_var_access(
-                    self.type_inf(expr), expr.name, None)
-
-    map_tagged_variable = map_variable
-
-    def map_subscript(self, expr):
-        return (self.count_var_access(self.type_inf(expr),
-                                      expr.aggregate.name,
-                                      expr.index)
-                + self.rec(expr.index))
-
-# }}}
-
+            # local memory access
+            local_mem_count_granularity = CountGranularity.SUBGROUP
 
-# {{{ GlobalMemAccessCounter
-
-class GlobalMemAccessCounter(MemAccessCounterBase):
-    def map_variable(self, expr):
-        name = expr.name
-
-        if name in self.knl.arg_dict:
-            array = self.knl.arg_dict[name]
-        else:
-            # this is a temporary variable
-            # FIXME temporary variable could have global address space
-            return self.new_zero_poly_map()
-
-        if not isinstance(array, lp.ArrayArg):
-            # this array is not in global memory
-            return self.new_zero_poly_map()
-
-        return self.new_poly_map({MemAccess(mtype="global",
-                    dtype=self.type_inf(expr), lid_strides={},
-                    gid_strides={}, variable=name,
+            if index is None:
+                return self.new_poly_map({MemAccess(
+                        address_space=AddressSpace.LOCAL,
+                        dtype=dtype,
+                        tags=tags,
+                        count_granularity=local_mem_count_granularity,
+                        kernel_name=self.knl.name): self.one})
+
+            # could be tuple or scalar index
+            index_tuple = index
+            if not isinstance(index_tuple, tuple):
+                index_tuple = (index_tuple,)
+
+            lid_strides, gid_strides = _get_lid_and_gid_strides(
+                                            self.knl, array, index_tuple)
+
+            return self.new_poly_map({MemAccess(
+                address_space=array.address_space,
+                dtype=dtype,
+                tags=tags,
+                lid_strides=immutabledict(lid_strides),
+                gid_strides=immutabledict(gid_strides),
+                variable=name,
+                count_granularity=local_mem_count_granularity,
+                kernel_name=self.knl.name): self.one})
+
+        elif (isinstance(array, TemporaryVariable) and (
```

Comment:
Should this also apply to `ArrayArg`? / Where do those get counted if not?

Reply:
Doesn't it also apply to `ArrayArg` (see 2 lines below)?